RELEASE-NOTES.md

```diff
@@ -1,6 +1,6 @@
 # Release Notes
 
-## PyMC 3.11.5 (TBD)
+## PyMC 3.11.5 (14 March 2022)
```

Comment:
Just noticed a typo below: https://github.com/pymc-devs/pymc/pull/5597/files#diff-7d8a453165f16d910a0937bf01782607cafc77fd7f42b443aff0cd02725848d7R8

Reply:
`pm.Lognormal` is now available as an alias for `pm.Lognormal` -> `pm.Lognormal` is now available as an alias for `pm.LogNormal`